pkg/registry/apis/provisioning/register.go

```diff
@@ -175,15 +176,110 @@ func RegisterAPIService(
+			}
+
+			switch a.GetResource() {
+			case provisioning.RepositoryResourceInfo.GetName():
+				// TODO: Support more fine-grained permissions than the basic roles. Especially on Enterprise.
+				switch a.GetSubresource() {
+				case "":
+					// Doing something with the repository itself.
+					if id.GetOrgRole().Includes(identity.RoleAdmin) {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "admin role is required", nil
+					}
+
+				case "test", "export", "migrate":
+					// Testing doesn't make sense for non-admins.
+					// Exporting and migrating are potentially dangerous.
+					if id.GetOrgRole().Includes(identity.RoleAdmin) {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "admin role is required", nil
+					}
+
+				case "webhook":
+					// When the resource is a webhook, we'll deal with permissions manually by checking signatures or similar in the webhook handler.
+					// The user in this context is usually an anonymous user, but may also be an authenticated synthetic check by the Grafana instance's operator as well.
+					// For context on the anonymous user, check the authn/clients/provisioning.go file.
+					return authorizer.DecisionAllow, "", nil
+
+				case "files":
+					// Reading files is allowed for everyone, so as to allow code reviews and similar.
+					// Writing files is only allowed fo Editor and higher.
+					isViewer := id.GetOrgRole().Includes(identity.RoleViewer)
+					isEditor := id.GetOrgRole().Includes(identity.RoleEditor)
+					isReadOperation := a.GetVerb() == apiutils.VerbGet || a.GetVerb() == apiutils.VerbList || a.GetVerb() == apiutils.VerbWatch
+
+					if isEditor || (isViewer && isReadOperation) {
+						return authorizer.DecisionAllow, "", nil
+					} else if isReadOperation {
+						return authorizer.DecisionDeny, "viewer role is required for reads", nil
+					} else {
+						return authorizer.DecisionDeny, "editor role is required for edits", nil
+					}
+
+				case "resources", "sync", "history", "render":
+					// These are strictly read operations.
+					// Sync can also be somewhat destructive, but it's expected to be fine to import changes.
+					if id.GetOrgRole().Includes(identity.RoleEditor) {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "editor role is required", nil
+					}
+
+				default:
+					if id.GetIsGrafanaAdmin() {
+						return authorizer.DecisionAllow, "", nil
+					} else {
+						return authorizer.DecisionDeny, "unmapped subresource defaults to no access", nil
+					}
+				}
+
+			case "stats":
+				// This can leak information one shouldn't necessarily have access to.
+				if id.GetOrgRole().Includes(identity.RoleAdmin) {
+					return authorizer.DecisionAllow, "", nil
+				} else {
+					return authorizer.DecisionDeny, "admin role is required", nil
+				}
```

Comment:
```suggestion
				if id.GetOrgRole().Includes(identity.RoleAdmin) {
					return authorizer.DecisionAllow, "", nil
				} 
				return authorizer.DecisionDeny, "admin role is required", nil
```

I don't have a strong opinion on this... but roberto usually asks me to change to this style